transfer/basic_download.go

```diff
@@ -203,7 +205,11 @@ func (a *basicDownloadAdapter) download(t *Transfer, cb TransferProgressCallback
 	}
 	written, err := tools.CopyWithCallback(dlFile, hasher, res.ContentLength, ccb)
 	if err != nil {
-		return fmt.Errorf("cannot write data to tempfile %q: %v", dlfilename, err)
+		wrappedErr := fmt.Errorf("cannot write data to tempfile %q: %v", dlfilename, err)
+		if errutil.IsRetriableError(err) {
+			return errutil.NewRetriableError(wrappedErr)
+		}
+		return wrappedErr
```

Comment:
After looking carefully at it I think your initial suggestion was correct. Please double check my error fix here: https://github.com/github/git-lfs/pull/1454/commits/a9418052bca0951916177f433ef1f5270380aff2

Other than that the PR is ready from my point of view 😄